lib/Onyx.js

```diff
@@ -1134,12 +1138,20 @@ function merge(key, changes) {
                 // We first only merge the changes, so we can provide these to the native implementation (SQLite uses only delta changes in "JSON_PATCH" to merge)
                 let batchedChanges = applyMerge(undefined, mergeQueue[key]);
 
+                if (_.isNull(batchedChanges)) {
+                    return remove(key);
+                }
+
+                // If the mergeQueue included a null value, we want to overwrite the existing value
+                // In this case, we can't simply merge the batched changes with the existing value, because then the null in the merge queue would have no effect
```

Comment:
Can you think of a cleaner way to do this besides checking for the existence of `null` inside the queued changes? 😄

Reply:
Not really.. 😄

The only other way i could think of is to move this logic to `applyMerge` and return an object like `{mergedChanges, shouldOverwriteKey}` or something. But then, this only makes sense for the `batchedChanges`, because the second call of `applyMerge` will never be `null`.

Reply:
Ok, I see. Just thinking out loud here - but let's say there is a `null` value in the merge queue... the value is either `null` or everything after a `null`? So wouldn't the "batched changes" essentially be the same as the existing value? Yet we merge them with `undefined`...

Could we also do this? 

```js
// The presence of a `null` in the merge queue instructs us to drop the existing value.
const shouldOverwriteExistingValue = _.includes(mergeQueue[key], null);
const updatedValue = shouldOverwriteExistingValue ? batchedChanges : applyMerge(existingValue, [batchedChanges]);
const modifiedData = utils.removeNullObjectValues(updatedValue);
```

Reply:
Yes, we can do that, you're right!